File: mobile/app/src/main/java/com/smartsuschef/mobile/data/models/WastageData.kt
package com.smartsuschef.mobile.data.models

data class WastageData(
    val id: String,
    val storeId: Int,
    val date: String,
    // Nullable (can be null if recipeId is set)
    val ingredientId: String?,
    // Nullable (can be null if ingredientId is set)
    val recipeId: String?,
    val quantity: Double,
    val createdAt: String,
    val updatedAt: String,
)

File: backend/SmartSusChef.Api/Models/WastageData.cs
namespace SmartSusChef.Api.Models;

public class WastageData
{
    public Guid Id { get; set; }
    public int StoreId { get; set; }
    public DateTime Date { get; set; }
    // Both are nullable so you can log one or the other
    public Guid? IngredientId { get; set; }
    public Guid? RecipeId { get; set; }
    public decimal Quantity { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Store Store { get; set; } = null!;

    // These MUST be nullable (?) because a specific record 
    // might not have one or the other.
    public Ingredient? Ingredient { get; set; } 
    public Recipe? Recipe { get; set; }
}
